File: frontend/src/types/course.ts
import type { Schedule } from '@/types/schedule'

export interface Course {
  id: string
  schedule_id: string
  date: string | null
  content: string | null
  follow_up: string | null

  schedule: Schedule | null
}

export interface CourseCreate {
  schedule_id: string
  date?: string
  content?: string
  follow_up?: string
}

export interface CourseUpdate {
  schedule_id?: string
  date?: string
  content?: string
  follow_up?: string
}

File: backend/app/models/course.py
from datetime import date
from typing import Optional

from pydantic import BaseModel
from app.models import Schedule

class Course(BaseModel):
    id: str = ""
    schedule_id: str = ""
    date: Optional[date]
    content: str = ""
    follow_up: str = ""

    #
    schedule: Optional[Schedule] = None

